### PR TITLE
Move pkg/scheduler to plugin/pkg/scheduler

### PR DESCRIPTION
**Pull request by hurf**
As the TODO in plugin/pkg/scheduler/scheduler.go described:

TODO: move everything from pkg/scheduler into this package. Remove
references from registry.

### CODE REVIEW

**Comment by hurf** [comment]
Strange, I have ran /hack/verify-gofmt.sh multi times locally, and it worked well.

**Comment by davidopp** [comment]
@abhgupta

File: plugin/pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -70,8 +68,8 @@ type Config struct {
 	// It is expected that changes made via modeler will be observed
 	// by MinionLister and Algorithm.
 	Modeler      SystemModeler
-	MinionLister scheduler.MinionLister
-	Algorithm    scheduler.Scheduler
+	MinionLister MinionLister
+	Algorithm    ScheduleAlgo
```
Comment (davidopp) [inline plugin/pkg/scheduler/scheduler.go]: Please call this SchedulerAlgorithm everywhere.

**Comment by abhgupta** [comment]
I am wondering if we need to add some structure to the predicates/priorities that we are moving over. 

We could perhaps have:
1. plugin/pkg/scheduler/algorithm/predicates/* --> contains predicates.go and any other files containing predicates
2. plugin/pkg/scheduler/algorithm/priorities/* --> contains priorities.go and the other files containing priorities
3. plugin/pkg/scheduler/algorithm/* --> contains the rest of the files from pkg/scheduler/*

Thoughts?

**Comment by hurf** [comment]
@abhgupta - sounds good, it clearly shows each component of a scheduler. I've modified the code as you said.

**Comment by abhgupta** [comment]
LGTM

**Comment by davidopp** [comment]
Can you please change the files you renamed to *iface* to *interface* ?

Other than that, LGTM. And good idea @abhgupta to split the files.

**Comment by hurf** [comment]
@davidopp - done.

**Comment by davidopp** [comment]
LGTM

**Comment by rjnagal** [comment]
@hurf Shippable failure looks legit. Can you take a look?

ok  	github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/scheduler/factory	0.093s	coverage: 76.9% of statements
panic: test timed out after 5m0s

goroutine 15 [running]:
testing.func·008()
	/usr/local/go/src/testing/testing.go:681 +0x168
created by time.goFunc
	/usr/local/go/src/time/sleep.go:129 +0x59

goroutine 1 [chan receive, 3 minutes]:
testing.RunTests(0x611818, 0x1029f80, 0x4, 0x4, 0xc208032101)
	/usr/local/go/src/testing/testing.go:556 +0xdc3
testing.(*M).Run(0xc20800a0f0, 0x1032bc0)
	/usr/local/go/src/testing/testing.go:485 +0xe1
main.main()
	github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait/_test/_testmain.go:106 +0x382

goroutine 11 [chan receive, 3 minutes]:
github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait.TestPollForever(0xc20805a120)
	/root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait/wait_test.go:128 +0x555
testing.tRunner(0xc20805a120, 0x1029fb0)
	/usr/local/go/src/testing/testing.go:447 +0x134
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:555 +0xd4e

goroutine 13 [chan send, 3 minutes]:
github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait.func·001()
	github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait/_test/_obj_test/wait.go:105 +0x24e
created by github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait.func·002
	github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait/_test/_obj_test/wait.go:112 +0x1a0
FAIL	github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait	300.018s
!!! Error in ./hack/test-go.sh:172
  'xargs -I{} -n1 -P${KUBE_COVERPROCS} go test "${goflags[@]:+${goflags[@]}}" ${KUBE_RACE} ${KUBE_TIMEOUT} -cover -covermode="${KUBE_COVERMODE}" -coverprofile="${cover_report_dir}/{}/${cover_profile}" "${cover_params[@]+${cover_params[@]}}" "${KUBE_GO_PACKAGE}/{}"' exited with status 0 123
Call stack:
  1: ./hack/test-go.sh:172 runTests(...)
  2: ./hack/test-go.sh:214 main(...)
Exiting with status 1

**Comment by hurf** [comment]
@rjnagal - I have some problem accessing shippable pages. It's stucked and showing only 'Waiting for build information...'.  From the log message you've posted, I didn't find any trace.

But I've modified the patch for several times, it's the first time shippable fails. And before that, I only changed two file names and already tested it in local environment. Let's wait for the result when Travis get well. If there're still errors, I can track them, is that okay?

**Comment by rjnagal** [comment]
Travis came back fine. Shippable might have data from last patch.

Merging, thanks.

**Pull request merged**